From ccc5b4ceb9fd590b5b245ab93da41415339d49e8 Mon Sep 17 00:00:00 2001
From: Chris Morgan <macromorgan@hotmail.com>
Date: Wed, 30 Oct 2024 20:59:43 -0500
Subject: [PATCH 07/10] arm64: dts: rockchip: Correctly identify always-on
 regulators for RGxx3

Identify which regulators can be shut off in suspend, and which
regulators aren't used at all. This was discovered through trial and
error as no schematics were available for any of the devices in
question.

 - regulator_vcc_wifi: I believe this can be disabled. Setting it to
   off in suspend has shown no ill effects.
 - vdd_logic: This is a critical regulator and cannot be shut off, even
   in suspend. Marking "regulator-on-in-suspend".
 - vdd_gpu: This regulator can be shut off in suspend with no observed
   ill effects.
 - LDO_REG2: I believe this regulator is unused.
 - LDO_REG9: I believe this regulator is unused.
 - BOOST: I am not yet sure if this regulator is used for the USB OTG
   port, but for now since we run the OTG port as a peripheral it is
   not used.
 - vdd_cpu: This regulator can be shut off in suspend with no observed
   ill effects.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 .../dts/rockchip/rk3566-anbernic-rgxx3.dtsi   | 31 +++++--------------
 1 file changed, 8 insertions(+), 23 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
index 1c2008efc5b8..494f4d38477a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi
@@ -186,10 +186,12 @@ vcc_wifi: regulator-vcc-wifi {
 		gpio = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
 		pinctrl-0 = <&vcc_wifi_h>;
 		pinctrl-names = "default";
-		regulator-always-on;
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 		regulator-name = "vcc_wifi";
+		regulator-state-mem {
+			regulator-off-in-suspend;
+		};
 	};
 
 	vibrator: pwm-vibrator {
@@ -289,7 +291,7 @@ vdd_logic: DCDC_REG1 {
 				regulator-initial-mode = <0x2>;
 				regulator-name = "vdd_logic";
 				regulator-state-mem {
-					regulator-off-in-suspend;
+					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <900000>;
 				};
 			};
@@ -342,15 +343,8 @@ regulator-state-mem {
 				};
 			};
 
-			vdda_0v9: LDO_REG2 {
-				regulator-always-on;
-				regulator-boot-on;
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <900000>;
-				regulator-name = "vdda_0v9";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+			LDO_REG2 {
+				/* unused */
 			};
 
 			vdda0v9_pmu: LDO_REG3 {
@@ -419,19 +413,11 @@ regulator-state-mem {
 				};
 			};
 
-			vcc2v8_dvp: LDO_REG9 {
-				regulator-always-on;
-				regulator-boot-on;
-				regulator-min-microvolt = <2800000>;
-				regulator-max-microvolt = <2800000>;
-				regulator-name = "vcc2v8_dvp";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+			LDO_REG9 {
+				/* Unused */
 			};
 
 			dcdc_boost: BOOST {
-				regulator-always-on;
 				regulator-min-microvolt = <4700000>;
 				regulator-max-microvolt = <5400000>;
 				regulator-name = "boost";
@@ -453,7 +439,6 @@ vdd_cpu: regulator@40 {
 		compatible = "fcs,fan53555";
 		reg = <0x40>;
 		fcs,suspend-voltage-selector = <1>;
-		regulator-always-on;
 		regulator-boot-on;
 		regulator-min-microvolt = <712500>;
 		regulator-max-microvolt = <1390000>;
-- 
2.43.0

